samtranslator/model/sam_resources.py

```diff
@@ -2237,6 +2245,93 @@ def _construct_appsync_api_resources(
 
         return api, cloudwatch_role
 
+    def _parse_auth_properties(self, api: GraphQLApi, auths: List[aws_serverless_graphqlapi.Auth]) -> None:
+        """
+        Parse the Auth properties in a Serverless::GraphQLApi resource.
+
+        This function does not return a value. The GraphQLApi object is modified directly so that we don't
+        return multiple config properties which are mostly None values.
+        """
+        if not auths:
+            raise InvalidResourceException(self.logical_id, "'Auth' must contain at least one valid authorizer.")
+
+        additional_auths: List[AdditionalAuthenticationProviderType] = []
+
+        # In each Auth index, you should only define a max of two properties. The "Type" property, and if
+        # necessary the associated config as well.
+        MAX_AUTH_PROPERTIES = 2
+
+        for index, auth in enumerate(auths):
+            # Check if more than one authentication config is defined, throw error if so
+            keys = remove_none_values(auth.dict()).keys()
+            if len(keys) > MAX_AUTH_PROPERTIES:
+                raise InvalidResourceException(
+                    self.logical_id, f"Index {index} in 'Auth' has more than one authentication configuration defined."
+                )
+
+            name, auth_dict = self._validate_auth_type_and_config(auth, index)
+
+            # The first index in authentication is the primary authentication
+            if index == 0:
+                api.AuthenticationType = auth.Type
+                if name:
+                    setattr(api, name, auth_dict)
+                continue
+
+            # All auths after the first are put into AdditionalAuthenticationProviders
+            additional_auth: AdditionalAuthenticationProviderType = {"AuthenticationType": auth.Type}
+            if name and auth_dict:
+                additional_auth[name] = auth_dict
+
+            additional_auths.append(additional_auth)
+
+        if additional_auths:
+            api.AdditionalAuthenticationProviders = additional_auths
```

Comment:
Just curious; what happens if you assign an empty list to `api.AdditionalAuthenticationProviders`?

Reply:
We'll create a property with empty value. Shouldn't be a big deal, but normally, it wouldn't exist if it's empty